internal/infrastructure/kubernetes/proxy/resource.go

```diff
@@ -371,6 +411,24 @@ func expectedContainerEnv(containerSpec *egv1a1.KubernetesContainerSpec) []corev
 			},
 		},
 	}
+	if gatewayNamespaceMode {
+		env = []corev1.EnvVar{
+			{
```

Comment:
curious whats this for ?

Reply:
This is setting an Envoy-gateway namespace envVar, which defaults to metadata.namespace field of envoy infra pod.